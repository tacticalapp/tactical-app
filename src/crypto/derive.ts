export async function derive(args: { data: string, salt: string }) {
    const ec = new TextEncoder();
    const key = await crypto.subtle.importKey(
        'raw',
        ec.encode(args.data.normalize('NFKD')),
        'PBKDF2',
        false,
        ['deriveBits']);
    const bits = await crypto.subtle.deriveBits({
        name: 'PBKDF2',
        hash: 'SHA-256',
        salt: ec.encode(args.salt),
        iterations: 100000,
    }, key, 256);
    return Buffer.from(bits);
}