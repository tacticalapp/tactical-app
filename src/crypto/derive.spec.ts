import { derive } from "./derive";

describe('derive', () => {
    it('should derive a key', async () => {
        let res = await derive({ data: 'password', salt: 'salt' });
        expect(res.toString('base64')).toMatchSnapshot();
    });
});