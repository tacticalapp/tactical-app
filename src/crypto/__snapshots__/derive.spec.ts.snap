// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`derive should derive a key 1`] = `"A5Si7eMyyaE+uC6bJGMWBMMd+Xi04vD70sVJlE+deaU="`;
